pkg/controller/daemon/daemon_controller.go

```diff
@@ -1341,11 +1356,47 @@ func NewPod(ds *apps.DaemonSet, nodeName string) *v1.Pod {
 	return newPod
 }
 
+// nodeSelectionPredicates runs a set of predicates that select candidate nodes for the DaemonSet.
+func nodeSelectionPredicates(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
```

Comment:
This predicate checks node name, node selector, node affinity specified in the pod spec. However, it's not obvious from the comment or function name. Would you make it more obvious?

Reply:
done.